graphite/graphite.go

```diff
@@ -100,6 +101,15 @@ func readTraceback(resp *http.Response) (*[]string, error) {
 // DefaultClient is the default HTTP client for requests.
 var DefaultClient = &http.Client{
 	Timeout: time.Minute,
+	Transport: &http.Transport{
+		MaxIdleConnsPerHost: 0,
```

Comment:
those issues are about the problem of http.Client returning errors that the caller of the http.Client shouldn't have to, or can't, deal with. (like here) and also http.Client not properly handling broken connections.

I haven't been able to consistently reproduce yet, but I indeed speculate that this solves the problem by just not trying to reuse connections.